src/test_l1_dsAudio.c

```diff
@@ -10213,55 +8795,45 @@ void test_l1_dsAudio_negative_dsGetSupportedARCTypes(void) {
  *
  */
 void test_l1_dsAudio_positive_dsAudioSetSAD(void) {
-	// Logging at the start
-	gTestID = 133;
-	UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
-
-	dsError_t result;
-	intptr_t* handle = NULL;
-	dsAudioSADList_t sadlist;
-	sadlist.sad[0] = 0; sadlist.sad[1] = 1; sadlist.sad[2] = 2; sadlist.sad[3] = 3;
-	sadlist.count = 4;
-
-        // allocate memory for the handles
-        handle = (intptr_t *) calloc(gDSAudioNumberOfPorts , sizeof(intptr_t));
-        if(handle == NULL)
-        {
-                UT_LOG_ERROR("Failed to allcoate memory for handlers");
-                return;
-        }
+    // Logging at the start
+    gTestID = 133;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
 
-	// Step 01: Initialize audio ports
-	result = dsAudioPortInit();
-	UT_ASSERT_EQUAL(result, dsERR_NONE);
+    dsError_t result;
+    intptr_t handle = null_handle;
+    dsAudioSADList_t sadlist;
+    sadlist.sad[0] = 0; sadlist.sad[1] = 1; sadlist.sad[2] = 2; sadlist.sad[3] = 3;
+    sadlist.count = 4;
 
-	// Step 02: Get the port handle for all supported audio ports
-	for (int i = 0; i < gDSAudioNumberOfPorts; i++) {
-                if(gDSAudioPortConfiguration[i].typeid != dsAUDIOPORT_TYPE_HDMI_ARC)
-                continue;
+    // Step 01: Initialize audio ports
+    result = dsAudioPortInit();
+    UT_ASSERT_EQUAL_FATAL(result, dsERR_NONE);
 
-		result = dsGetAudioPort(gDSAudioPortConfiguration[i].typeid, gDSAudioPortConfiguration[i].index, (handle+i));
-		UT_ASSERT_EQUAL(result, dsERR_NONE);
-		UT_ASSERT_NOT_EQUAL(*(handle+i), null_handle);
+    if(gSourceType == 0) 
+    {
+            // Step 02: Get the port handle for all supported audio ports
+            for (int i = 0; i < gDSAudioNumberOfPorts; i++) {
+                    if(gDSAudioPortConfiguration[i].typeid != dsAUDIOPORT_TYPE_HDMI_ARC)
+                          continue;
 
-		// Step 03: Set SAD for HDMI ARC port
-                result = dsAudioSetSAD(*(handle+i), sadlist);
-                if(gSourceType == 0) {
+                    result = dsGetAudioPort(gDSAudioPortConfiguration[i].typeid, gDSAudioPortConfiguration[i].index, &handle);
                     UT_ASSERT_EQUAL(result, dsERR_NONE);
-                } else {
-                    UT_ASSERT_EQUAL(result, dsERR_OPERATION_NOT_SUPPORTED);
-                }
-	}
+                    UT_ASSERT_NOT_EQUAL(handle, null_handle);
 
-	// Step 04: Terminate audio ports
-	result = dsAudioPortTerm();
-	UT_ASSERT_EQUAL(result, dsERR_NONE);
+                    // Step 03: Set SAD for HDMI ARC port
+                    result = dsAudioSetSAD(handle, sadlist);
+           }        UT_ASSERT_EQUAL(result, dsERR_NONE);
+    } else {
+            UT_ASSERT_EQUAL(result, dsERR_OPERATION_NOT_SUPPORTED);
```

Comment:
comparison is wrong, call setSAD and then compare the result

Reply:
done